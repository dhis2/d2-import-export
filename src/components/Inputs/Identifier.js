import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { identifierOptions } from '../../utils/options'
import { RadioGroupField } from '../'

const NAME = 'identifier'
const DATATEST = 'input-identifier'
const LABEL = i18n.t('Identifier')

const Identifier = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={identifierOptions}
        dataTest={DATATEST}
    />
)

export { Identifier }
