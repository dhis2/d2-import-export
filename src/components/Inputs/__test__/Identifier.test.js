import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Identifier } from '../Identifier'

jest.mock('react-final-form')

describe('Input component - Identifier', () => {
    it('should render correctly', () => {
        const file = shallow(<Identifier />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
