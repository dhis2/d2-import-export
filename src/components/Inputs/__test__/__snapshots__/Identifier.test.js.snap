// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Identifier should render correctly 1`] = `
<RadioGroup
  dataTest="input-identifier"
  label="Identifier"
  name="identifier"
  options={
    Array [
      Object {
        "label": "UID",
        "value": "UID",
      },
      Object {
        "label": "Code",
        "value": "CODE",
      },
      Object {
        "label": "Auto",
        "value": "AUTO",
      },
    ]
  }
/>
`;
